src/preprocess.c

```diff
@@ -0,0 +1,338 @@
+/*
+ * Created by Yotam Levit on 24/07/2024.
+*/
+
+#include "../include/logger.h"
+#include "../include/preprocess.h"
+#include "../include/utils.h"
+#include <stdio.h>
+#include <string.h>
+
+
+#define MACRO_START "macr"
+#define MACRO_END "endmacr\n"
+#define EOS '\0'
+#define NEW_LINE '\n'
```

Comment:
Please consider moving lines 14 - 15 to the global file, I assume other modules will use them as well